src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue29421.cs

```diff
@@ -0,0 +1,24 @@
+#if TEST_FAILS_ON_WINDOWS // Related issue for windows: https://github.com/dotnet/maui/issues/29245
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+public class Issue29421 : _IssuesUITest
+{
+    public override string Issue => "KeepScrollOffset Not Working as Expected in CarouselView";
+
+    public Issue29421(TestDevice device)
+    : base(device)
+    { }
+
+    [Test]
+    [Category(UITestCategories.CarouselView)]
+    public void VerifyCarouselViewKeepScrollOffsetAdd()
+    {
+        App.WaitForElement("carouselview");
+        App.Tap("AddButton");
+        VerifyScreenshot();
```

Comment:
Pending snapshot on Mac.
![image](https://github.com/user-attachments/assets/7c1fbe23-6ac8-419a-9fb9-60253c0c3de6)
Could you commit it?